kernel/ap_start/src/lib.rs

```diff
@@ -27,10 +31,13 @@ pub static AP_READY_FLAG: AtomicBool = AtomicBool::new(false);
 static AP_STACKS: MutexIrqSafe<BTreeMap<u32, NoDrop<Stack>>> = MutexIrqSafe::new(BTreeMap::new());
 
 /// Insert a new stack that was allocated for the AP with the given `cpu_id`.
-pub fn insert_ap_stack(cpu_id: u32 , stack: Stack) {
-    AP_STACKS.lock().insert(cpu_id, NoDrop::new(stack));
+pub fn insert_ap_stack(cpu_id: u32, stack: NoDrop<Stack>) {
+    AP_STACKS.lock().insert(cpu_id, stack);
```

Comment:
is there any reason why we force the caller to deal with `NoDrop` now rather than doing it within this function?

Reply:
Not anymore :ok_hand: